cloud/secret_new.go

```diff
@@ -86,6 +85,17 @@ func (c *client) GetUserSecret(ctx context.Context, secretName string) (*Secret,
 	return c.getSecretV2(ctx, fmt.Sprintf("/user/%s", secretName))
 }
 
+func (c *client) GetUserOrProjectSecret(ctx context.Context, path string) (*Secret, error) {
```

Comment:
Mind adding a comment?
Would `GetSecret` be a fitting name too?

Reply:
> Would GetSecret be a fitting name too?

I want to make it clear that this is for the new project-based secrets and not the original secrets implementation.

Maybe when we fully remove support for the old secrets, we could then simplify these names.